subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -2407,6 +2416,14 @@ static inline int rx_demux_rx(memq_link_t *link, struct node_rx_hdr *rx)
 		ull_scan_aux_release(link, rx);
 	}
 	break;
+#if defined(CONFIG_BT_CTLR_SYNC_PERIODIC)
+	case NODE_RX_TYPE_SYNC_REPORT:
+	{
+		(void)memq_dequeue(memq_ull_rx.tail, &memq_ull_rx.head, NULL);
+		ull_sync_report(link, rx);
+	}
+	break;
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
```

Comment:
`ull_sync_report`  will check `ftr->sync_status` for every `NODE_RX_TYPE_SYNC_REPORT`, this beats the purpose of adding `lll_sync_create_prepare`. Instead, why not prepare and send `NODE_RX_TYPE_SYNC` from  the `lll_sync_create_prepare` control path?

Reply:
I've changed approach. Now The NODE_RX_TYPE_SYNC is used to transport information about sync established and first received PDU. That is handled by `ull_sync_established_report` (I've renamed the function) function. NODE_RX_TYPE_SYNC_REPORT are handled as they were in the past.